### PR TITLE
Ingest ssSource designation data from the LSST alert stream

### PR DESCRIPTION
**Pull request by thomasculino**
This pull request adds support for Rubin/LSST Solar System object (SSO) metadata in alert handling, specifically by introducing the `SsSource` struct to represent per-detection SSO ephemeris and designation information. The changes ensure that SSO data is parsed, stored, indexed, and exposed throughout the alert ingestion, enrichment, and filtering pipelines, and are accompanied by new tests for correct deserialization and handling of SSO fields.

**Support for Solar System Object metadata:**

- Added the new `SsSource` struct to represent Solar System object ephemeris and designation data, with full parsing and (de)serialization support in `src/alert/lsst.rs`. This struct is now included in the `LsstRawAvroAlert`, `LsstAlert`, and `LsstObject` types, and is handled during alert ingestion and enrichment. 
- The SSO designation is now persisted in the auxiliary collection for each object, updated independently of cross-survey matches, and indexed for efficient lookup by designation. 

**Pipeline and API changes:**

- The alert enrichment and filtering pipelines have been updated to propagate the new `ss_source` field, making SSO metadata available for downstream consumers and APIs. 

**Testing and validation:**

- Added tests to verify correct deserialization of the new `SsSource` struct, including edge cases where the field is absent. Existing tests and fixtures were updated to include or expect the new fields where appropriate. 

These changes collectively enable robust handling and querying of Solar System object information in LSST alert data, paving the way for more advanced moving object science use cases.

The idea behind this PR is to query by `ssObjectId` in the future

### CODE REVIEW

File: src/alert/lsst.rs [hunk 1]
```diff
@@ -1282,6 +1407,22 @@ impl AlertWorker for LsstAlertWorker {
...
+        // The designation is only known once a diaSource is linked to an ssObject; keep it
+        // current on the aux doc independent of whichever branch above ran, and regardless of
+        // any ZTF cross-match (an LSST-only or ZTF-only ssObject is a normal outcome).
+        if ss_source_present {
+            self.alert_aux_collection
+                .update_one(
+                    doc! { "_id": &object_id },
+                    match &designation {
+                        Some(designation) => doc! { "$set": { "designation": designation } },
+                        None => doc! { "$unset": { "designation": "" } },
```
Comment (mcoughlin) [inline src/alert/lsst.rs]: Probably should ensure that once an MPC association is made, not to unset it if a future alert doesn't have it for some reason?

File: src/alert/lsst.rs [hunk 2]
```diff
@@ -1282,6 +1407,22 @@ impl AlertWorker for LsstAlertWorker {
             }
         }
 
+        // The designation is only known once a diaSource is linked to an ssObject; keep it
+        // current on the aux doc independent of whichever branch above ran, and regardless of
+        // any ZTF cross-match (an LSST-only or ZTF-only ssObject is a normal outcome).
+        if ss_source_present {
+            self.alert_aux_collection
```
Comment (mcoughlin) [inline src/alert/lsst.rs:1452]: Could this go into the existing aux insert/update?

File: src/utils/db.rs [hunk 3]
```diff
@@ -103,12 +103,19 @@ pub async fn initialize_survey_indexes(
...
     create_index(&alerts_collection, index, false).await?;
 
-    // if survey is LSST, create an index on the ss_object_id field of the alerts collection
+    // if survey is LSST, create an index on the ssObjectId field of the alerts collection,
+    // and on the designation field of the aux collection (used to look up a moving object by
+    // its MPC designation, independent of any cross-survey position match)
     if survey == &Survey::Lsst {
         let index = doc! {
-            "ss_object_id": 1,
+            "ssObjectId": 1,
```
Comment (mcoughlin) [inline src/utils/db.rs:130]: Can this be a partial index? { designation: { $exists: true } } as most objects won't have ssObjectId?

**Review by mcoughlin — COMMENTED** [review]
Looking really good!

**Review by mcoughlin — APPROVED** [review]
LGTM

**Review by mcoughlin — APPROVED** [review]
Let's goooooo.
